lib/sequenced/generator.rb

```diff
@@ -23,8 +23,10 @@ def skip?
     end
 
     def next_id
-      next_id_in_sequence.tap do |id|
-        id += 1 until unique?(id)
+      record.class.with_table_lock do
+        next_id_in_sequence.tap do |id|
+          id += 1 until unique?(id)
+        end
       end
```

Comment:
This doesn't guarantee race-free behavior as far as I can tell.

Consider this scenario:

Thread 1 locks the table and uses this method to get the next id in the sequence.
Thread 1 releases the lock.
 - CONTEXT SWITCH -
Thread 2  locks the table and uses this method to get the next id in the sequence.
Thread 2 releases the lock.
Thread 2 inserts successfully.
- CONTEXT SWITCH - 
Thread 1 attempts to insert with the same value that thread 2 just used.

Result: disaster.

See my PR here for the correct place to put the lock https://github.com/djreimer/sequenced/pull/16 .